### DISCUSSION TITLE
Errors with pip-compile

### DISCUSSION THREAD

**Discussion by mjhuber** [renovatebot/renovate#22139] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

34.13.2

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

15.4

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

We have a renovate config:
```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "labels": [
    "renovate"
  ],
  "pip-compile": {
    "fileMatch": [
      "(^|/)requirements\\.in$",
      "/pypi/requirements_lock.in",
      "/^pypi\/requirements_lock.in$/gm",
      "pypi/requirements_lock.in"
    ]
  },
  "packageRules": [
    {
      "matchUpdateTypes": [
        "patch"
      ],
      "enabled": true
    },
    {
      "matchUpdateTypes": [
        "major,minor"
      ],
      "enabled": true,
      "extends": [
        "schedule:monthly"
      ]
    }
  ],
  "prConcurrentLimit": 4
}
```

and we'd like to use pip-compile to update `pypi/requirements_lock.in` and build the deps.  Here's our `requirements_lock.in`:
```
#  Run the below command to keep requirements_lock.txt up to date
#
#    pur -r pypi/requirements_lock.in && sort-requirements pypi/requirements_lock.in && pip-compile --resolver=backtracking pypi/requirements_lock.in
#
absl-py==1.4.0
gitpython==3.1.31
google-cloud-storage==2.9.0
marshmallow==3.19.0
mysql-replication==0.40
netaddr==0.8.0
packaging==23.1
psycopg==3.1.9
py3dns==3.2.0
pymysql==1.0.3
python-dateutil==2.8.2
python-dotenv==1.0.0
pytz==2023.3
pyyaml==6.0
requests==2.30.0
semver==3.0.0
slackclient==2.9.4
smart-open[gcs]==6.3.0
switch==1.1.0
tqdm==4.65.0
```

Renovate can find the updates just find but isn't able to compile the deps and we get an error message in the PR that gets created:
```
Command failed: install-tool python 3.11.3
/opt/buildpack/tools/python/3.11.3/bin/python: error while loading shared libraries: libpython3.11.so.1.0: cannot open shared object file: No such file or directory
```

Any suggestions what we can do to get this working?

### What you tried so far.

currently using this config:
```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "labels": [
    "renovate"
  ],
  "pip-compile": {
    "fileMatch": [
      "(^|/)requirements\\.in$",
      "/pypi/requirements_lock.in",
      "/^pypi\/requirements_lock.in$/gm",
      "pypi/requirements_lock.in"
    ]
  },
  "packageRules": [
    {
      "matchUpdateTypes": [
        "patch"
      ],
      "enabled": true
    },
    {
      "matchUpdateTypes": [
        "major,minor"
      ],
      "enabled": true,
      "extends": [
        "schedule:monthly"
      ]
    }
  ],
  "prConcurrentLimit": 4
}
```

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by viceice** [comment]
I'm not sure about the error but your `fileMatch` is wrong.

`/pypi/requirements_lock.in` will never match
`/^pypi\/requirements_lock.in$/gm` flags are not supported, so this will never match anything.

**Comment by HonkingGoose** [comment]
> ## If you're self-hosting Renovate, tell us what version of Renovate you run.
> 
> 34.13.2

That version is from 2022-11-02. Also, there are _newer_ updates in the `34.x` release.

I recommend you at least update to the latest version of `34.x`, and if possible update to the latest version of `35.x`.
